src/components/MouseAwayListener.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { ReactElement, Ref, SyntheticEvent, useEffect, useRef } from "react";
+import { isChildOf } from "../utils/reactUtils";
+
+export interface IMouseAwayListenerProps {
+  children?: ReactElement<{ ref: Ref<HTMLElement> }>;
+  onMouseAway?(e: SyntheticEvent): void;
+}
+
+export default function MouseAwayListener({ children, onMouseAway }: IMouseAwayListenerProps) {
```

Comment:
😶‍🌫️ 🫠 😵‍💫 nice! That just makes me think why React requires all of these for a pop over.

Reply:
yeah it's a bit insane lol. we have a second code location where we could use this (T1MenuItem) b/c you'll actually notice the options menu icon still being visible after everything closes